BrainPortal/app/views/userfiles/show.html.erb

```diff
@@ -31,6 +31,14 @@
   <% if @userfile.can_be_accessed_by?(current_user, requested_access = :read) %>
     <%= link_to(@userfile.is_a?(FileCollection) ? "Download Collection" : "Download File", { :action => 'download', :file_ids => [@userfile.id] }, :class => "button menu_button", :method  => :get) %>
   <% end %>
+
+  <% if @userfile.can_be_accessed_by?(current_user, requested_access = :write) && @userfile.is_a?(SingleFile) %>
```

Comment:
Make the compress/uncompress button work for both SingleFile and FileCollection (resp: compression vs archiving). This is what the file manager interface already does.